### PR TITLE
Genre should maintain case

### PR DESCRIPTION
**Pull request by pgwillia**
In coordination with #559, if you edit a Genre field with some invalid term
or alternative capitalization like "documentary" and save the form your
response will be accepted and not challenged. Investigation showed that
the validation does occur, but when there is alternative capitalization
it is accepted by the validation.  This change corrects the validation.

#545

### CODE REVIEW

File: app/models/genre_term.rb [hunk 1]
```diff
@@ -55,7 +55,7 @@ def autocomplete(query)
     end
 
     def normalize_text(text)
-      text.downcase.gsub(/[^a-zA-Z0-9_]+/, '_')
+      text.gsub(/[^a-zA-Z0-9_]+/, '_')
```
Comment (pgwillia) [inline app/models/genre_term.rb]: https://github.com/ualbertalib/avalon/blob/fbbf8c790ad4148e8b557bc4f69be95301cdfd26/app/models/media_object.rb#L67
https://github.com/ualbertalib/avalon/blob/fbbf8c790ad4148e8b557bc4f69be95301cdfd26/app/models/media_object.rb#L121-L125
https://github.com/ualbertalib/avalon/blob/fbbf8c790ad4148e8b557bc4f69be95301cdfd26/app/models/genre_term.rb#L36-L44
https://github.com/ualbertalib/avalon/blob/fbbf8c790ad4148e8b557bc4f69be95301cdfd26/app/models/genre_term.rb#L29-L34

**Comment by pgwillia** [comment]
> +1

Thanks Jeff, but the "Correctly handle genre" test is failing.  Maybe this isn't ready.

**Comment by jefferya** [comment]
> > +1
> 
> Thanks Jeff, but the "Correctly handle genre" test is failing. Maybe this isn't ready.

Saw this after the fact and testing in the browser.
Inspected the test. The test was added by #573 and it assumed the previous functionality of Genre. I.e., if a lowercase genre reached the model, it would match the id (instead of the text) in GenreTerm and after the reload the genre wouldn't have the lowercase version.

https://github.com/ualbertalib/avalon/pull/573/files#diff-8b68c60904c2054f0ebdf1b50ffdb642R431-R435

**Comment by jefferya** [comment]
Looking at the failing test. Can confirm he test assumes the previous implementation of `normalize_text`

https://github.com/ualbertalib/avalon/blob/avalon-6/app/models/genre_term.rb#L58

The test assume that the `text` would be downcase thus `Interview` and `interview` would match in `find`(and return the associated `GenreTerm` item)

https://github.com/ualbertalib/avalon/blob/avalon-6/app/models/genre_term.rb#L29-L34

Thus without downcase, `find` fails for for `interview` thus causing the test to fail. To fix, my proposal, remove the `interview` downcase version from the following and replace with a test for `validate_genre`
  
https://github.com/ualbertalib/avalon/blob/avalon-6/spec/models/media_object_spec.rb#L431

https://github.com/ualbertalib/avalon/blob/avalon-6/spec/models/media_object_spec.rb#L435

For validation, `language` might be an ok pattern:

https://github.com/ualbertalib/avalon/blob/avalon-6/spec/models/media_object_spec.rb#L37-L47

**Comment by jefferya** [comment]
:+1:

**Pull request merged**